crates/router/src/connector/rapyd.rs

```diff
@@ -0,0 +1,663 @@
+mod transformers;
+use std::fmt::Debug;
+
+use base64::Engine;
+use bytes::Bytes;
+use common_utils::date_time;
+use error_stack::{IntoReport, ResultExt};
+use rand::distributions::{Alphanumeric, DistString};
+use ring::hmac;
+use transformers as rapyd;
+
+use crate::{
+    configs::settings,
+    consts,
+    core::{
+        errors::{self, CustomResult},
+        payments,
+    },
+    headers, logger, services,
+    types::{self, api, ErrorResponse, Response},
+    utils::{self, BytesExt},
+};
+
+#[derive(Debug, Clone)]
+pub struct Rapyd;
+
+impl Rapyd {
+    pub fn generate_signature(
+        &self,
+        auth: &rapyd::RapydAuthType,
+        http_method: &str,
+        url_path: &str,
+        body: &str,
+        timestamp: &i64,
+        salt: &str,
+    ) -> CustomResult<String, errors::ConnectorError> {
+        let rapyd::RapydAuthType {
+            access_key,
+            secret_key,
+        } = auth;
+        let to_sign =
+            format!("{http_method}{url_path}{salt}{timestamp}{access_key}{secret_key}{body}");
+        let key = hmac::Key::new(hmac::HMAC_SHA256, secret_key.as_bytes());
+        let tag = hmac::sign(&key, to_sign.as_bytes());
+        let hmac_sign = hex::encode(tag);
+        let signature_value = consts::BASE64_ENGINE_URL_SAFE.encode(hmac_sign);
+        Ok(signature_value)
+    }
+}
+
+impl api::ConnectorCommon for Rapyd {
+    fn id(&self) -> &'static str {
+        "rapyd"
+    }
+
+    fn common_get_content_type(&self) -> &'static str {
+        "application/json"
+    }
+
+    fn base_url<'a>(&self, connectors: &'a settings::Connectors) -> &'a str {
+        connectors.rapyd.base_url.as_ref()
+    }
+
+    fn get_auth_header(
+        &self,
+        _auth_type: &types::ConnectorAuthType,
+    ) -> CustomResult<Vec<(String, String)>, errors::ConnectorError> {
+        Ok(vec![])
+    }
+}
+
+impl api::PaymentAuthorize for Rapyd {}
+
+impl
+    services::ConnectorIntegration<
+        api::Authorize,
+        types::PaymentsAuthorizeData,
+        types::PaymentsResponseData,
+    > for Rapyd
+{
+    fn get_headers(
+        &self,
+        _req: &types::PaymentsAuthorizeRouterData,
+        _connectors: &settings::Connectors,
+    ) -> CustomResult<Vec<(String, String)>, errors::ConnectorError> {
+        Ok(vec![])
```

Comment:
is the headers are empty?

Reply:
have common headers(eg: Content_type, etc) here, and call the `get_headers` in `build_request` and append the extra signature related once to it.